frontend/src/pages/notebookController/screens/server/EnvironmentVariablesField.tsx

```diff
@@ -39,8 +39,9 @@ const EnvironmentVariablesField: React.FC<EnvironmentVariablesFieldProps> = ({
   }, [variable.type]);
 
   const handleSecretChange = (checked: boolean) => {
-    variable.type = checked ? 'password' : 'text';
-    setVariableType(variable.type);
+    const updatedVariable = { ...variable };
+    updatedVariable.type = checked ? 'password' : 'text';
+    setVariableType(updatedVariable.type);
```

Comment:
This is a change in logic. Likely needs to call `onUpdateVariable`.

The simplest solution today would be to disable the rule here and add a TODO comment.